### PR TITLE
feat: add getFeatureVariableJSON and getAllFeatureVariables apis

### PR DESCRIPTION
**Pull request by mnoman09**
## Summary
- add new APIs (getFeatureVariableJSON, getAllFeatureVariables)
- add new notification type (all-feature-variables)
- Add mavenLocal() to supported repositories for local development
- Bump core SDK to 3.5.0-beta

- Added unit tests

### CODE REVIEW

File: android-sdk/src/main/java/com/optimizely/ab/android/sdk/OptimizelyClient.java [hunk 1]
```diff
@@ -671,6 +672,98 @@ String getFeatureVariableString(@NonNull String featureKey,
...
+     * @param variableKey The unique key of the variable.
+     * @param userId      The ID of the user.
+     * @return An OptimizelyJSON instance for the JSON variable value.
+     * Null if the feature or variable could not be found.
+     */
+    @Nullable
+    public OptimizelyJSON getFeatureVariableJSON(@NonNull String featureKey,
+                                                 @NonNull String variableKey,
+                                                 @NonNull String userId) {
+        if (isValid()) {
```
Comment (msohailhussain) [inline android-sdk/src/main/java/com/optimizely/ab/android/sdk/OptimizelyClient.java:688]: We don't really need to call in every method.

Comment (msohailhussain) [reply android-sdk/src/main/java/com/optimizely/ab/android/sdk/OptimizelyClient.java:688]: Call `getFeatureVariableJSON` of this class with default values. same for getAllFeatureVariables

Comment (msohailhussain) [reply android-sdk/src/main/java/com/optimizely/ab/android/sdk/OptimizelyClient.java:688]: invalid comment. nvm.

File: build.gradle [hunk 2]
```diff
@@ -45,6 +45,7 @@ allprojects {
     repositories {
         jcenter()
         google()
+        mavenLocal()
```
Comment (mikeng13) [inline build.gradle:48]: This should be commented out and only enabled manually by whoever is running locally. Just leave a comment in there about it

**Pull request merged**